interfaces/builtin/gpg_public_keys.go

```diff
@@ -0,0 +1,57 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const gpgPublicKeysSummary = `allows reading gpg public keys and non-sensitive configuration`
+
+const gpgPublicKeysBaseDeclarationSlots = `
+  gpg-public-keys:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const gpgPublicKeysConnectedPlugAppArmor = `
+# Description: Can read gpg public keys and non-sensitive configuration
+
+/usr/share/gnupg/options.skel r,
+owner @{HOME}/.gnupg/ r,
+owner @{HOME}/.gnupg/gpg.conf r,
+owner @{HOME}/.gnupg/pubring.gpg{,.lock} r,
+owner @{HOME}/.gnupg/pubring.gpg.lock k,
```

Comment:
These are all locks with read, which means we only allow opens with O_RDONLY and shared/read locks (eg LOCK_SH). The 'r' (O_RDONLY) with an exclusive/write lock (eg LOCK_EX) requires 'w' with the 'k'. As such, these rules do allow locking writes but not reads. I recall adding these since the man page referenced them, but neither gpg1 or gpg2 needs them for encrypt/verify/list/etc so I'll remove them.

Reply:
This is done.